### ISSUE TITLE
Improve the section about Single Root Element

### ISSUE THREAD

**Issue by Masa-Shin** [vuejs/vuejs.org#2147]
https://vuejs.org/v2/guide/components.html#A-Single-Root-Element

> So this might be a good time to refactor the <blog-post> component to accept a single post prop instead:

I suppose refactoring to accept a single prop is not a good practice now for the following reasons.

1. You can now pass an object to the component without changing the definition of the properties, as mentioned here: https://vuejs.org/v2/guide/components-props.html#Passing-the-Properties-of-an-Object

2. It makes harder to validate each original props.

So I'm wondering if we should replace it with one using v-bind without an argument (or at least add a note about it).

**Comment by phanan** [comment]
Sounds good! What do you think about something like this, which should show both ways to achieve what we want, and still is concise enough?

```html
<blog-post
  v-for="post in posts"
  v-bind:key="post.id"
  v-bind:post="post" <!-- or v-bind="post" -->
></blog-post>
```

**Comment by Masa-Shin** [comment]
Thank you! Looks fine but the latter syntax would be invalid after refactoring to accept a single prop, so maybe a bit misleading I think. Could we put a small note like '!) We also supported [Passing the Properties of an Object](https://vuejs.org/v2/guide/components-props.html#Passing-the-Properties-of-an-Object)' or something like that if space allows.

**Comment by sarahdayan** [comment]
Do you think it's necessary to have this section about refactoring many props to a single one undr [A Single Root Element](https://vuejs.org/v2/guide/components.html#A-Single-Root-Element)? It feels like it doesn't really belong here.

Besides, the [Passing Data to Child Components with Props](https://vuejs.org/v2/guide/components.html#Passing-Data-to-Child-Components-with-Props) section right before already starts tackling that topic. It feels like it definitely would belong there instead. What do you think?

**Comment by Masa-Shin** [comment]
I agree with @sarahdayan. It is about passing properties and fits better in the previous section.

**Comment by Masa-Shin** [comment]
The [Single Root Element] (https://vuejs.org/v2/guide/components.html#A-Single-Root-Element) section needs rewrite a bit for consistency then.
